client/failover.go

```diff
@@ -0,0 +1,91 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/drand/drand/beacon"
+	"github.com/drand/drand/key"
+	"github.com/drand/drand/log"
+)
+
+const defaultFailoverGracePeriod = time.Second * 5
+
+// NewFailoverWatcher creates a client whose Watch function will failover to
+// Get-ing new randomness if it does not receive it after the passed grace period.
+func NewFailoverWatcher(core Client, group *key.Group, gracePeriod time.Duration, l log.Logger) Client {
+	if gracePeriod == 0 {
+		gracePeriod = defaultFailoverGracePeriod
+	}
+
+	return &failoverWatcher{
+		Client:      core,
+		group:       group,
+		gracePeriod: gracePeriod,
+		log:         l,
+	}
+}
+
+type failoverWatcher struct {
+	Client
+	group       *key.Group
+	gracePeriod time.Duration
+	log         log.Logger
+}
+
+// Watch returns new randomness as it becomes available.
```

Comment:
When the failover logic switches between GETting and receiving from WATCH, it can result in sending duplicate and/or reordered rounds to its user.

Reply:
We keep track of the latest round so they’ll never be duplicated or reordered, only potentially missed.

Reply:
My bad. You are right. Rounds can be skipped, but not duplicated or reordered. That sounds good.